client-data/js/board.js

```diff
@@ -436,6 +435,7 @@ var scaleTimeout = null;
 Tools.setScale = function setScale(scale) {
 	if (isNaN(scale)) scale = 1;
 	scale = Math.max(0.1, Math.min(10, scale));
+	scale = Math.min(scale, document.documentElement.clientWidth/Tools.server_config.MAX_BOARD_SIZE_X);
```

Comment:
We could just add this to the Math.min above

```suggestion
	var max_scale = document.documentElement.clientWidth/Tools.server_config.MAX_BOARD_SIZE_X;
	scale = Math.max(0.1, Math.min(10, max_scale, scale));
```